core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -236,8 +236,16 @@ protected[core] object WhiskWebActionsApi extends Directives {
           // the following throws an exception if the code is
           // not a whole number or a valid code
           StatusCode.int2StatusCode(c.toIntExact)
-
-        case _ => throw new Throwable("Illegal code")
+        case JsString(c) =>
+          try {
+            val intCode = c.toInt
```

Comment:
FWIW the try could have been simply around `c.toInt` since `int2StatusCode` will throw an exception anyway. Since neither of the throw exceptions are observable, it's OK to keep it as is.